File: src/main/java/com/sber/yamlprocessor/model/ProcessConfig.java
package com.sber.yamlprocessor.model;

import com.fasterxml.jackson.annotation.JsonProperty;
import jakarta.validation.Valid;
import jakarta.validation.constraints.NotNull;

public class ProcessConfig {
    
    @NotNull(message = "Process is required")
    @Valid
    private Process process;

    public ProcessConfig() {}

    public ProcessConfig(Process process) {
        this.process = process;
    }

    public Process getProcess() {
        return process;
    }

    public void setProcess(Process process) {
        this.process = process;
    }
}

File: src/main/kotlin/com/sber/yamlprocessor/model/ProcessConfig.kt
package com.sber.yamlprocessor.model

import jakarta.persistence.CascadeType
import jakarta.persistence.Entity
import jakarta.persistence.OneToOne
import jakarta.persistence.Table

@Entity
@Table(name = "process_config")
class ProcessConfig(
    @OneToOne(mappedBy = "processConfig", cascade = [CascadeType.ALL], orphanRemoval = true)
    var process: Process? = null
) : BaseEntity()
